pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,167 @@
+package org.jboss.pnc.bacon.pig.impl.addons.scanservice;
+
+import org.jboss.pnc.bacon.config.Validate;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+
+class ValidationTest {
+
+    PostBuildScanService scanService = new PostBuildScanService(null, null, null, null);
+
+    private static final Map<String, ?> benchmarkScanServiceConfigMap = new LinkedHashMap<>();
+    static {
+        benchmarkScanServiceConfigMap.put("productID", null);
+        benchmarkScanServiceConfigMap.put("eventId", null);
+        benchmarkScanServiceConfigMap.put("isManagedService", null);
+        benchmarkScanServiceConfigMap.put("cpaasVersion", null);
+        benchmarkScanServiceConfigMap.put("jobUrl", null);
+        benchmarkScanServiceConfigMap.put("serviceUrl", null);
+        benchmarkScanServiceConfigMap.put("serviceSecretKey", null);
+        benchmarkScanServiceConfigMap.put("serviceSecretValue", null);
+        benchmarkScanServiceConfigMap.put("brewBuilds", null);
+        benchmarkScanServiceConfigMap.put("scmUrl", null);
+        benchmarkScanServiceConfigMap.put("scmRevision", null);
+        benchmarkScanServiceConfigMap.put("extraScmUrls", null);
+        benchmarkScanServiceConfigMap.put("extraBuilds", null);
+        benchmarkScanServiceConfigMap.put("name", null);
+    }
+
+    private static final Map<String, ?> minimalConfigMap = new LinkedHashMap<>();
+    static {
+        minimalConfigMap.put("productID", null);
+        minimalConfigMap.put("serviceUrl", null);
+        minimalConfigMap.put("serviceSecretKey", null);
+        minimalConfigMap.put("serviceSecretValue", null);
+    }
+
+    @Test
+    void positiveKeywordsTest() {
+        showMap(benchmarkScanServiceConfigMap);
+        try {
+            scanService.validateMapKeys(benchmarkScanServiceConfigMap);
+            Assertions.assertTrue(true);
+        } catch (Exception e) {
+            Assertions.fail();
```

Comment:
As with other assertion usage - won't the test implicitly fail if an exception is thrown and therefore this all seems redundant?

Reply:
We have two similar tests: positive and negative. Reverse logic applies for negative tests when validation procedure discover an invalid key and throws an exception. This denotes the correct behaviour and assert(true) is called from a catch block whereas assert(false) is called from the try block to indicate failure. Because of that we cannot rely on implicit fail in both cases. To keep it consistent we use asserts in both cases even though positive test case could rely on implicit fail as you pointed out.